test/sde/sde_dynamical.jl

```diff
@@ -31,17 +40,116 @@ end
     g_iip(du,u,p,t) = du .= g(u,p,t)
 
     ff_harmonic = DynamicalSDEFunction(f1_harmonic,f2_harmonic,g)
-    prob1 = DynamicalSDEProblem(ff_harmonic,v0,u0,(0.0,5.0))
-    sol1 = solve(prob1,BAOAB(gamma=γ);dt=1/10,save_noise=true)
+    prob1 = DynamicalSDEProblem(ff_harmonic,g,v0,u0,(0.0,5.0))
+    sol1 = solve(prob1,BAOAB(gamma=[γ,γ]);dt=1/10,save_noise=true)
 
     prob2 = DynamicalSDEProblem(f1_harmonic_iip,f2_harmonic_iip,g_iip,v0,u0,(0.0,5.0); noise=NoiseWrapper(sol1.W))
-    sol2 = solve(prob2,BAOAB(gamma=γ);dt=1/10)
+    sol2 = solve(prob2,BAOAB(gamma=[γ,γ]);dt=1/10)
 
     @test sol1[:] ≈ sol2[:]
 
     dts = (1/2) .^ (8:-1:4)
 
-    # Can't use NoiseGrid as noise is not generated with the correct size in convergence.jl. We require noise with shape of v.
-    sim1  = analyticless_test_convergence(dts,prob1,BAOAB(gamma=γ),(1/2)^10;trajectories=Int(1e2),use_noise_grid=false)
-    @test abs(sim1.𝒪est[:weak_final]-1) < 0.3
+    # # Can't use NoiseGrid as noise is not generated with the correct size in convergence.jl. We require noise with shape of v.
+    # sim1  = analyticless_test_convergence(dts,prob1,BAOAB(gamma=γ),(1/2)^10;trajectories=Int(1e2),use_noise_grid=false)
+    # @test abs(sim1.𝒪est[:weak_final]-1.5) < 0.3
+
+
+    sol1 = solve(prob1,ABOBA(gamma=[γ,γ]);dt=1/10,save_noise=true)
+    prob2 = DynamicalSDEProblem(f1_harmonic_iip,f2_harmonic_iip,g_iip,v0,u0,(0.0,5.0); noise=NoiseWrapper(sol1.W))
+    sol2 = solve(prob2,ABOBA(gamma=[γ,γ]);dt=1/10)
+
+    @test sol1[:] ≈ sol2[:]
+
+    # # Can't use NoiseGrid as noise is not generated with the correct size in convergence.jl. We require noise with shape of v.
+    # sim1  = analyticless_test_convergence(dts,prob1,ABOBA(gamma=γ),(1/2)^10;trajectories=Int(1e4),use_noise_grid=false)
+    # @test abs(sim1.𝒪est[:weak_final]-2) < 0.3
+
+
+    sol1 = solve(prob1,OBABO(gamma=[γ,γ]);dt=1/10,save_noise=true)
+    prob2 = DynamicalSDEProblem(f1_harmonic_iip,f2_harmonic_iip,g_iip,v0,u0,(0.0,5.0); noise=NoiseWrapper(sol1.W))
+    sol2 = solve(prob2,OBABO(gamma=[γ,γ]);dt=1/10)
+
+    @test sol1[:] ≈ sol2[:]
+
+    # # Can't use NoiseGrid as noise is not generated with the correct size in convergence.jl. We require noise with shape of v.
```

Comment:
Can you not predefine a NoiseGrid with the correct size?

Reply:
This comment comes from the previous implementation of BAOAB #397 as the tests are more or less the same and were copied up for ABOBA and OBABO, so I don't really know and I did not try myself, but I left the comment